src/oc_erchef/apps/chef_objects/src/chef_node.erl

```diff
@@ -157,9 +159,16 @@ ejson_for_indexing(#chef_node{name = Name, environment = Environment}, Node) ->
                                    %% FIXME: nodes may have environment in the db, but not in JSON
                                    %% or not set at all (pre-environments nodes).
                                    {<<"chef_environment">>, Environment},
-                                   {<<"recipe">>, extract_recipes(RunList)},
-                                   {<<"role">>, extract_roles(RunList)},
-                                   {<<"run_list">>, RunList}]),
+                                   {<<"recipe">>, ExtractedRecipes},
+                                   {<<"role">>, ExtractedRoles},
+                                   {<<"run_list">>, RunList},
+                                   {<<"cheftoplevel-name">>, Name},
+                                   {<<"cheftoplevel-chef_environment">>, Environment},
+                                   {<<"cheftoplevel-recipe">>, ExtractedRecipes},
+                                   {<<"cheftoplevel-role">>, ExtractedRoles},
+                                   {<<"cheftoplevel-run_list">>, RunList},
+                                   {<<"cheftoplevel-recipes">>, ej:get({"recipes"}, Automatic)},
```

Comment:
What will the result here be if "recipes" doesn't exist in the node json?  This is the sort of case that would be good to have a test around so that the behavior doesn't change in the future.